pyproject.toml

```diff
@@ -141,6 +141,9 @@ indent-style = "space"
 docstring-quotes = "double"
 inline-quotes = "single"
 
+[tool.ruff.lint.flake8-type-checking]
+runtime-evaluated-base-classes = ["pydantic.BaseModel"]
+
```

Comment:
Am I correct in understanding that with this setting, we should be able to use type fields for Pydantic models within `TYPE_CHECKING`? Or what is this good for? If we are gonna use it, please update it in the Crawlee & Client as well.

Reply:
Without this, ruff is complaining that annotations used in Pydantic classes are not conditionally imported with **if TYPE_CHECKING**. This tells ruff to not complain about it as those annotations are needed on runtime and not only during static code analysis.

Reply:
I will do small PRs for other repos to include it there as well.